### PR TITLE
bump msal, add unsubscribe

### PR DESCRIPTION
**Pull request by derisen**
# CHANGELOG

## 10/02/2020

* Updated MSAL.js to 1.4.1 (resolves id token cache issue).
* Added unsubscribe to broadcast events on ngOnDestroy().

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -22,7 +27,7 @@ export class AppComponent implements OnInit {
     this.checkAccount();
 
     // event listeners for authentication status
-    this.broadcastService.subscribe('msal:loginSuccess', (success) => {
+    this.subscription1 = this.broadcastService.subscribe('msal:loginSuccess', (success) => {
```
Comment (jasonnutter) [inline src/app/app.component.ts:30]: I would call these `loginSuccessSubscription`, loginFailureSubscription`, respetively. And they can be local variables, dont need to be added as instance properties.

File: src/app/app.component.ts [hunk 2]
```diff
@@ -69,6 +74,15 @@ export class AppComponent implements OnInit {
       correlationId: CryptoUtils.createNewGuid(),
       piiLoggingEnabled: false
     }));
+
+    this.subscriptions.push(this.subscription1);
+    this.subscriptions.push(this.subscription2);
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscriptions.length > 0) {
```
Comment (jasonnutter) [inline src/app/app.component.ts:83]: Don't need to check length when using `forEach`, sice it is initialized to an empty array.

Comment (derisen) [reply src/app/app.component.ts]: aah, of course.

Comment (derisen) [reply src/app/app.component.ts]: done!

File: src/app/app.component.ts (hunk 1 above)
Comment (derisen) [reply src/app/app.component.ts]: done!

**Comment by derisen** [comment]
Thanks for the review! Merging.

**Pull request merged**